precompile/contracts/rewardmanager/contract_test.go

```diff
@@ -164,11 +269,60 @@ var (
 				require.Equal(t, constants.BlackholeAddr, address)
 			},
 		},
+		"no log disable rewards if DUpgrade is not active": {
+			Caller:     allowlist.TestManagerAddr,
+			BeforeHook: allowlist.SetDefaultRoles(Module.Address),
+			InputFn: func(t testing.TB) []byte {
+				input, err := PackDisableRewards()
+				require.NoError(t, err)
+
+				return input
+			},
+			ChainConfigFn: func(ctrl *gomock.Controller) precompileconfig.ChainConfig {
+				mockChainConfig := precompileconfig.NewMockChainConfig(ctrl)
+				mockChainConfig.EXPECT().GetFeeConfig().AnyTimes().Return(commontype.ValidTestFeeConfig)
+				mockChainConfig.EXPECT().AllowedFeeRecipients().AnyTimes().Return(false)
+				mockChainConfig.EXPECT().IsDUpgrade(gomock.Any()).AnyTimes().Return(false)
+				return mockChainConfig
+			},
+			SuppliedGas: SetRewardAddressGasCost,
+			ReadOnly:    false,
+			ExpectedRes: []byte{},
+			AfterHook: func(t testing.TB, stateDB contract.StateDB) {
+				// Check logs are not stored in state
+				topics, data := stateDB.GetLogData()
+				require.Len(t, topics, 0)
+				require.Len(t, data, 0)
+			},
+		},
+		"log disable rewards if DUpgrade is active": {
```

Comment:
For this pair of tests and the other new ones, aren't these creating duplicate tests since there were tests for the pre DUpgrade behavior before (which have been modified to include the event, but not including a check for the event) ?

Could we consolidate these tests and make sure that the names are clear? I think something like `disable rewards pre / post DUpgrade` would make it a bit more clear and also make it more obvious if we've accidentally created duplicates

Reply:
By default all tests run with DUpgrade activated. I think this is the default behaviour in most of our tests (to run with latest chain config). If that's the case I don't think we can rename every one of them with post DUpgrade. 

The reason I duplicated the tests and named them with pre/post Dupgrade was to follow append-like fashion to add tests rather than change targets of existing tests. 

The question here should be: If there was something wrong with events, do we want to fail the existing tests as well or just the new ones? BTW I'm totally fine to add duplicated test setup (unless they also test the same outcome).

Reply:
I see, I hadn't thought of it that way.

Curious what @marun recommends for testing best practices as well.

I'll defer to you guys on the best way to handle these go-forward tests.

Reply:
I think most of the tests are upgraded through `TestChainConfig` which activates the latest upgrade. But we don't remove/deprecate any of tests that target specific upgrades.

The case here is simply having two tests that have two exact same setups and steps, but testing different outcomes. One of them tests if the caller is allowed to call the function, the other tests if the outcome of the call has correct outputs.

Reply:
Yes, the VM must be able to re-process any historical block that was processed with a previous rule set.

Reply:
I refactored those by adding new assert helper functions for events and used them in existing tests.